src/Libraries/Microsoft.Extensions.AI.Abstractions/SpeechToText/SpeechToTextClientExtensions.cs

```diff
@@ -0,0 +1,116 @@
+    /// <returns>The text generated by the client.</returns>
+    public static Task<SpeechToTextResponse> GetResponseAsync(
+        this ISpeechToTextClient client,
+        Stream speechStream,
+        SpeechToTextOptions? options = null,
+        CancellationToken cancellationToken = default)
+        => Throw.IfNull(client)
+            .GetResponseAsync(
+                [speechStream.ToAsyncEnumerable(cancellationToken: cancellationToken)],
+                options,
+                cancellationToken);
+
+    /// <summary>Generates text from speech providing a single speech audio <see cref="DataContent"/>.</summary>
+    /// <param name="client">The client.</param>
+    /// <param name="speechStream">The single speech audio stream.</param>
+    /// <param name="options">The speech to text options to configure the request.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The text generated by the client.</returns>
+    public static IAsyncEnumerable<SpeechToTextResponseUpdate> GetStreamingResponseAsync(
+        this ISpeechToTextClient client,
+        Stream speechStream,
+        SpeechToTextOptions? options = null,
+        CancellationToken cancellationToken = default)
+        => Throw.IfNull(client)
+            .GetStreamingResponseAsync(
+                [speechStream.ToAsyncEnumerable(cancellationToken: cancellationToken)],
+                options,
+                cancellationToken);
+
+    /// <summary>Generates text from speech providing a single speech audio <see cref="DataContent"/>.</summary>
+    /// <param name="client">The client.</param>
+    /// <param name="speechContent">The single speech audio content.</param>
+    /// <param name="options">The speech to text options to configure the request.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The text generated by the client.</returns>
+    public static IAsyncEnumerable<SpeechToTextResponseUpdate> GetStreamingResponseAsync(
+        this ISpeechToTextClient client,
+        DataContent speechContent,
+        SpeechToTextOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        IEnumerable<DataContent> speechContents = [Throw.IfNull(speechContent)];
+        return Throw.IfNull(client)
+            .GetStreamingResponseAsync(
+                [speechContents.ToAsyncEnumerable()],
+                options,
+                cancellationToken);
+    }
+
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    private static async IAsyncEnumerable<T> ToAsyncEnumerable<T>(this IEnumerable<T> source)
```

Comment:
Eventually we'll be able to replace this with the one in System.Linq.AsyncEnumerable, but we'll need to wait I expect until that's out of preview